lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
I thought a bit more about this now after viewing your diff. I think we should keep that how it is for now and here is why:

We use the model as "dispatchable" for all other entities to, so the models contain some federation methods (`subscribers` and `public?`), so the same object is used as model and for dispatching. It's the same for all entities and retraction is no "special case" here.

The only thing special about retractions is, that it's not a database model, it has only a short lifespan and the data is stored in redis instead of postgres/mysql. So the difference is only in `Workers::DeferredRetraction` where the "dispatchable" isn't loaded from the database, but after that everything works the same as for all other entities.

So if we want to split federation logic from the models, we should do that for all entities, and that is in my eyes completely out of scope for this PR. And even then I'm not sure if the overhead of many more classes is worth it or if we should just keep it as it is, since for example `public?` can be used directly from the model now for some entities.